state-chain/pallets/cf-funding/src/lib.rs

```
@@ -900,6 +918,56 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		#[pallet::call_index(12)]
+		#[allow(clippy::let_unit_value)]
+		#[pallet::weight(Weight::zero())]
+		pub fn execute_sc_call(
+			origin: OriginFor<T>,
+			deposit_and_call: EthereumDepositAndSCCall,
+			caller: EthereumAddress,
+			caller_account_id: AccountId<T>,
+			// Required to ensure this call is unique per funding event.
+			tx_hash: EthTransactionHash,
+		) -> DispatchResult {
+			T::EnsureWitnessed::ensure_origin(origin)?;
+
+			// process the deposit
+			match deposit_and_call.deposit {
+				EthereumDeposit::FlipToSCGateway { amount } =>
+					Self::fund_sc_account(caller_account_id.clone(), caller, amount.into(), tx_hash),
+
+				// Deposit via vault or transfers will be handled here in the future
+				_ => {},
+			}
+
+			match EthereumSCApi::decode(&mut &deposit_and_call.call[..]) {
+				Ok(call) => {
+					// If the call fails to execute, we still succeed this extrinsic since we need
+					// to successfully process the deposit above. In this case, the deposit
+					// will be processed and no call will be executed.
+					match call
+						.clone()
+						.dispatch_bypass_filter(RuntimeOrigin::<T>::signed(caller_account_id))
+					{
+						Ok(_) => {
+							Self::deposit_event(Event::SCCallExecuted { sc_call: call, tx_hash });
+						},
+						Err(e) => {
+							log::warn!(
+								"SC call couldn't be executed. It returned an error: {:?}",
```

We could maybe also return error here or fire an event with the error that indicates that the execution has failed instead of just swallow it and only log a warn.

---

This was on purpose. We cant return an error because it will fail the extrinsic and the storages would revert. We dont want that. Even if the call fails, we want the deposit to be processed as stated in the comment above.

---

done.